src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,194 @@
+                && conditionalAccessExpressionSyntax.WhenNotNull.IsKind(SyntaxKind.InvocationExpression);
+
+            return expressionNode.IsKind(SyntaxKind.InvocationExpression)
+                   || isNullConditionalInvocationExpression
+                   || expressionNode.IsKind(SyntaxKind.ObjectCreationExpression)
+                   || expressionNode is AssignmentExpressionSyntax
+                   || expressionNode.IsKind(SyntaxKind.PreIncrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PreDecrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PostIncrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.PostDecrementExpression)
+                   || expressionNode.IsKind(SyntaxKind.AwaitExpression);
+        }
+
+        protected override bool TryGetInlineSyntaxNodeAndReplacementNodeForDelegate(
+            InvocationExpressionSyntax calleeInvocationNode,
+            IMethodSymbol calleeMethodSymbol,
+            ExpressionSyntax inlineExpressionNode,
+            StatementSyntax statementContainsCallee,
+            SyntaxGenerator syntaxGenerator,
+            out SyntaxNode? inlineSyntaxNode,
+            out SyntaxNode? syntaxNodeToReplace)
+        {
+            inlineSyntaxNode = null;
+            syntaxNodeToReplace = null;
+            if (statementContainsCallee is LocalDeclarationStatementSyntax localDeclarationSyntax
+                && IsVariableInitializerInLocalDeclarationSyntax(calleeInvocationNode, localDeclarationSyntax)
+                && localDeclarationSyntax.Declaration.Type.IsVar)
+            {
+                // Example:
+                // Before:
+                // void Caller() { var x = Callee(); }
+                // Action Callee() { return () => {}; }
+                //
+                // After inline it should be
+                // void Caller() { Action x = () => {};}
+                // Action Callee() { return () => {}; }
+                // 'var' can't be used for delegate
+                inlineSyntaxNode = UseExplicitTypeAndReplaceInitializerForDeclarationSyntax(
+                    localDeclarationSyntax,
+                    syntaxGenerator,
+                    calleeMethodSymbol.ReturnType,
+                    calleeInvocationNode,
+                    inlineExpressionNode);
+
+                syntaxNodeToReplace = statementContainsCallee;
+                return true;
+            }
+
+            // Example:
+            // Before:
+            // void Caller() { var x = Callee()(); }
+            // Func<int> Callee() { return () => 1; }
+            // After:
+            // void Caller() { var x = ((Func<int>)(() => 1))(); }
+            // Func<int> Callee() { return () => 1; }
+            // Cast expression is needed for lambda
+            if (calleeInvocationNode.Parent?.IsKind(SyntaxKind.InvocationExpression) == true)
```

Comment:
this cast should likely be done unilaterally as long as the inlining point is not an expression-statement.  wherever the method body is otherwise inlined into we should cast to preserve semantics.  teh cast will be removed if not necessary later by the simplifier.